src/conversions/std/num.rs

```diff
@@ -106,15 +107,19 @@ macro_rules! int_fits_c_long {
 
         impl<'source> FromPyObject<'source> for $rust_type {
             fn extract(obj: &'source PyAny) -> PyResult<Self> {
-                let ptr = obj.as_ptr();
-                let val = unsafe {
-                    let num = ffi::PyNumber_Index(ptr);
-                    if num.is_null() {
-                        Err(PyErr::fetch(obj.py()))
-                    } else {
-                        let val = err_if_invalid_value(obj.py(), -1, ffi::PyLong_AsLong(num));
-                        ffi::Py_DECREF(num);
-                        val
+                // fast path - checking for subclass of `int` just checks a bit in the type object
+                let val = if let Ok(long) = obj.downcast::<PyLong>() {
+                    err_if_invalid_value(obj.py(), -1, unsafe { ffi::PyLong_AsLong(long.as_ptr()) })
```

Comment:
I admit that I don't really understand why we're using `PyLong_AsLong`, not `PyLong_AsLongLong` here? But I kept it as-is.

Reply:
Note the name of the macro this is part of: `int_fits_c_long`. Types which could benefit from a larger representation use the other macros defined below.

Reply:
Makes sense, I got confused as wikipedia says `c long` is 32bits, but I think that's mostly not the case.

Reply:
On most platforms `long` is the "native" width (so 64-bit on 64-bit); on Windows it's always 32-bit.